Classes/Audio/VorbisSpectrogramGenerator.cs

```diff
@@ -72,29 +113,62 @@ public class VorbisSpectrogramGenerator {
         }
 
         var fftSize = (int)Math.Pow(2, Editor.Spectrogram.FftSizeExp);
-        var sg = new SpectrogramGenerator(sampleRate, fftSize: fftSize, stepSize: Editor.Spectrogram.StepSize, maxFreq: Editor.Spectrogram.MaxFreq);//, fixedWidth: (int)height);
-        sg.Colormap = Colormap.Blues;
+        var sg = new SpectrogramGenerator(sampleRate, fftSize: fftSize, stepSize: Editor.Spectrogram.StepSize, maxFreq: maxFreq);
+        sg.Colormap = Colormap.GetColormap(colormap);
         sg.Add(audioBufferDouble);
 
-        Bitmap bmp = sg.GetBitmapMel(melBinCount: Editor.Spectrogram.Width);
-        //Bitmap bmp = sg.GetBitmap();
-        //bmp.Save("halMel.png", ImageFormat.Png);
+        Bitmap bmp = null;
+        switch (type) {
+            case SpectrogramType.Standard:
+                bmp = sg.GetBitmap();
+                break;
+            case SpectrogramType.MelScale:
+                bmp = sg.GetBitmapMel(melBinCount: Editor.Spectrogram.MelBinCount);
+                break;
+            case SpectrogramType.MaxScale:
+                bmp = sg.GetBitmapMax();
+                break;
+        }
+        if (cache) {
+            bmp.Save(cachedBmpSpectrogramPath, ImageFormat.Png);
```

Comment:
With current settings, the cached images are quite large for longer songs (~9 MB) and since all selected possibilities are cached, the storage racks up quickly.

Investigate if reducing the horizontal resolution still gives ok results - maybe add an option for the user to reduce quality to speed up the generation and reduce storage space?

Reply:
Previous step size gives very good quality of the image - I've set that as High quality.

Increasing the step size by a factor of 2 reduced the quality only slightly, almost not noticeably - I've set that as Medium quality and a default option.

Increasing the step size by a factor of 4 reduced the quality significantly - I've set that as Low quality option.